pkg/app/piped/executor/kubernetes/kubernetes.go

```diff
@@ -212,11 +212,28 @@ func applyManifests(ctx context.Context, applier provider.Applier, manifests []p
 		lp.Infof("Start applying %d manifests to %q namespace", len(manifests), namespace)
 	}
 	for _, m := range manifests {
-		if err := applier.ApplyManifest(ctx, m); err != nil {
-			lp.Errorf("Failed to apply manifest: %s (%v)", m.Key.ReadableLogString(), err)
-			return err
+		annotation := m.GetAnnotations()[provider.LabelSyncReplace]
+		// Do not replace CRD because replace might delete CRD instance
+		if annotation == provider.UserReplaceTrue && !m.Key.IsCRD() {
+			// Always try to replace first and create if it fails due to resource not found error.
+			// This is because we cannot know whether resource already exists before executing command.
+			err := applier.ReplaceManifest(ctx, m)
+			if errors.Is(err, provider.ErrNotFound) {
+				lp.Infof("Specified resource does not exist, so create the resource: %s (%w)", m.Key.ReadableLogString(), err)
+				err = applier.CreateManifest(ctx, m)
+			}
+			if err != nil {
+				lp.Errorf("Failed to replace or create manifest: %s (%w)", m.Key.ReadableLogString(), err)
+				return err
+			}
+			lp.Successf("- replaced or created manifest: %s", m.Key.ReadableLogString())
+		} else {
+			if err := applier.ApplyManifest(ctx, m); err != nil {
+				lp.Errorf("Failed to apply manifest: %s (%w)", m.Key.ReadableLogString(), err)
+				return err
+			}
+			lp.Successf("- applied manifest: %s", m.Key.ReadableLogString())
 		}
```

Comment:
[nits] you can write without if-else but it's up to you.
```suggestion
                if annotation != provider.UserReplaceTrue || m.Key.IsCRD() {
	                if err := applier.ApplyManifest(ctx, m); err != nil {
		                lp.Errorf("Failed to apply manifest: %s (%w)", m.Key.ReadableLogString(), err)
		                return err
	                }
	                lp.Successf("- applied manifest: %s", m.Key.ReadableLogString())
	                continue
                }
                // Always try to replace first and create if it fails due to resource not found error.
                // This is because we cannot know whether resource already exists before executing command.
                err := applier.ReplaceManifest(ctx, m)
                if errors.Is(err, provider.ErrNotFound) {
	                lp.Infof("Specified resource does not exist, so create the resource: %s (%w)", m.Key.ReadableLogString(), err)
	                err = applier.CreateManifest(ctx, m)
                }
                if err != nil {
	                lp.Errorf("Failed to replace or create manifest: %s (%w)", m.Key.ReadableLogString(), err)
	                return err
                }
                lp.Successf("- replaced or created manifest: %s", m.Key.ReadableLogString())

```

Reply:
Thank you ! 
I think this is better too